Jenkinsfile

```diff
@@ -38,10 +45,32 @@ pipeline {
       }
     }
 
-    stage ('Deploy') {
+    stage ('Deploy Snapshot') {
       when {
         expression {
-          return env.BRANCH_NAME ==~ /(^[0-9]+$)|(^(([0-9]+)(\.))+([0-9]+)?$)|(^master$)/
+          return env.BRANCH_NAME ==~ /(^master$)/ && VERSION ==~ /.*-SNAPSHOT$/
+        }
+      }
+      steps {
+        script {
+          rtMaven.deployer.deployArtifacts buildInfo
+          server.publishBuildInfo buildInfo
+        }
+      }
+      post {
+        failure {
+          rocketSend channel: 'jenkinsohsome', message: "Deployment of oshdb-build nr. ${env.BUILD_NUMBER} *failed* on Branch - ${env.BRANCH_NAME}  (<${env.BUILD_URL}|Open Build in Jenkins>). Latest commit from  ${author}. Is Artifactory running?" , rawMessage: true
+        }
+      }
+    }
+
+    stage ('Deploy Release') {
+      when {
+        allOf {
+          tag pattern: "(^[0-9]+\$)|(^(([0-9]+)(\\.))+([0-9]+)?\$)", comparator: "REGEXP"
```

Comment:
Maybe we could tweak the regex a bit, to allow beta/RC releases? And I believe we would never do a major-version only release like `7`, so the first part of the regex could be dropped.

From what I can [see](https://docs.oracle.com/middleware/1212/core/MAVEN/maven_version.htm#MAVEN400), maven uses versions in the style of `<major version>.<minor version>.<incremental version>-<qualifier>`, where all parts are numbers, except for the qualifier which can be a string and the SNAPSHOT qualifier has some special meaning… I guess if we could limit ourselves to a fixed set of qualifiers (`alpha-[0-9]+`, `beta-[0-9]+`, `RC[0-9]+`).

Reply:
How about: `/^([0-9]+(\.[0-9]+)*?)(-(RC|beta-|alpha-)[0-9]+)?$/` ([regex101](https://regex101.com/r/7pFHIE/1))

Or better: `/^([0-9]+(\.[0-9]+)*)(-(RC|beta|alpha)-?[0-9]+)?$/` ([regex101](https://regex101.com/r/XJVDCz/1))

Reply:
let's use the first one

Reply:
right, so I use `RELEASE_REGEX = /^([0-9]+(\.[0-9]+)*)(-(RC|beta-|alpha-)[0-9]+)?$/`

Reply:
right, so I use `RELEASE_REGEX = /^([0-9]+(\.[0-9]+)*)(-(RC|beta-|alpha-)[0-9]+)?$/` for both, tag name and version number